server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotLoader.java

```diff
@@ -217,142 +224,203 @@ private void deleteAllFilesInDataDirs() throws IOException {
 
   private void createLinksFromSnapshotDirToDataDirWithoutLog(File sourceDir)
       throws IOException, DiskSpaceInsufficientException {
-    File seqFileDir = new File(sourceDir, "sequence" + File.separator + storageGroupName);
-    File unseqFileDir = new File(sourceDir, "unsequence" + File.separator + storageGroupName);
+    File seqFileDir =
+        new File(
+            sourceDir,
+            IoTDBConstant.SEQUENCE_FLODER_NAME
+                + File.separator
+                + storageGroupName
+                + File.separator
+                + dataRegionId);
+    File unseqFileDir =
+        new File(
+            sourceDir,
+            IoTDBConstant.UNSEQUENCE_FLODER_NAME
+                + File.separator
+                + storageGroupName
+                + File.separator
+                + dataRegionId);
     if (!seqFileDir.exists() && !unseqFileDir.exists()) {
       throw new IOException(
           String.format(
               "Cannot find %s or %s",
               seqFileDir.getAbsolutePath(), unseqFileDir.getAbsolutePath()));
     }
-
-    File[] seqRegionDirs = seqFileDir.listFiles();
-    if (seqRegionDirs != null && seqRegionDirs.length > 0) {
-      for (File seqRegionDir : seqRegionDirs) {
-        if (!seqRegionDir.isDirectory()) {
-          LOGGER.info("Skip {}, because it is not a directory", seqRegionDir);
+    folderManager =
```

Comment:
Initializing the `folderManager` outside but using it inside several methods is wired.